src/helpers/userHelper.js

```diff
@@ -1049,34 +1049,36 @@ const userHelper = function () {
 
   // 'Most Hrs in Week'
   const checkMostHrsWeek = async function (personId, user, badgeCollection) {
-    let badgeOfType;
-    for (let i = 0; i < badgeCollection.length; i += 1) {
-      if (badgeCollection[i].badge?.type === 'Most Hrs in Week') {
-        badgeOfType = badgeCollection[i].badge;
-      }
-    }
-    await badge.findOne({ type: 'Most Hrs in Week' }).then((results) => {
-      userProfile
-        .aggregate([
-          { $match: { isActive: true } },
-          { $group: { _id: 1, maxHours: { $max: '$lastWeekTangibleHrs' } } },
-        ])
-        .then((userResults) => {
-          if (
-            user.lastWeekTangibleHrs
-            && user.lastWeekTangibleHrs >= userResults[0].maxHours
-          ) {
-            if (badgeOfType) {
-              increaseBadgeCount(
-                personId,
-                mongoose.Types.ObjectId(badgeOfType._id),
-              );
-            } else {
-              addBadge(personId, mongoose.Types.ObjectId(results._id));
+    if (
+      user.weeklycommittedHours > 0 &&
+      user.lastWeekTangibleHrs > user.weeklycommittedHours
+    ) {
+      const badgeOfType = badgeCollection
+        .filter((object) => object.badge.type === "Minimum Hours Multiple")
+        .map((object) => object.badge);
+      await badge.findOne({ type: 'Most Hrs in Week' }).then((results) => {
+        userProfile
+          .aggregate([
+            { $match: { isActive: true } },
+            { $group: { _id: 1, maxHours: { $max: '$lastWeekTangibleHrs' } } },
+          ])
+          .then((userResults) => {
+            if (
+              user.lastWeekTangibleHrs
+              && user.lastWeekTangibleHrs >= userResults[0].maxHours
+            ) {
+              if (badgeOfType) {
```

Comment:
I suggest changing this to check that the length of badgeOfType is greater than 0. When I first tested badgeOfType = [ ] and it was increasing count every time instead of adding a new badge.